contracts/hydro/src/contract.rs

```diff
@@ -269,26 +285,7 @@ pub fn execute(
         ExecuteMsg::RemoveAccountFromWhitelist { address } => {
             remove_from_whitelist(deps, env, info, address)
         }
-        ExecuteMsg::UpdateConfig {
-            activate_at,
-            max_locked_tokens,
-            known_users_cap,
-            max_deployment_duration,
-            cw721_collection_info,
-            lock_depth_limit,
-            lock_expiry_duration_seconds,
-        } => update_config(
-            deps,
-            env,
-            info,
-            activate_at,
-            max_locked_tokens,
-            known_users_cap,
-            max_deployment_duration,
-            cw721_collection_info,
-            lock_depth_limit,
-            lock_expiry_duration_seconds,
-        ),
+        ExecuteMsg::UpdateConfig { config } => update_config(deps, env, info, config),
```

Comment:
def good change here. how does this impact the validation? can you leave fields of the config empty?

Reply:
Yes, all fields except `activate_at` are made optional, so you can choose which ones you want to update.

```
pub struct UpdateConfigData {
    pub activate_at: Timestamp,
    pub max_locked_tokens: Option<u128>,
    pub known_users_cap: Option<u128>,
    pub max_deployment_duration: Option<u64>,
    pub cw721_collection_info: Option<CollectionInfo>,
    pub lock_depth_limit: Option<u64>,
    pub lock_expiry_duration_seconds: Option<u64>,
    pub slash_percentage_threshold: Option<Decimal>,
    pub slash_tokens_receiver_addr: Option<String>,
}
```